src/queue.c

```diff
@@ -195,6 +195,30 @@ mpd_send_add_id_to(struct mpd_connection *connection, const char *uri,
 	return mpd_send_s_u_command(connection, "addid", uri, to);
 }
 
+bool
+mpd_send_add_id_whence(struct mpd_connection *connection, const char *uri,
+		   unsigned to, enum mpd_position_whence whence)
+{
+	const char *whence_s = "";
+	switch (whence) {
+	case MPD_POSITION_ABSOLUTE:
+		break;
+
+	case MPD_POSITION_AFTER_CURRENT:
+		whence_s = "+";
+		break;
+
+	case MPD_POSITION_BEFORE_CURRENT:
+		whence_s = "-";
+		break;
+	}
+
+	char to_str[64] = "";
+	snprintf(to_str, 64, " position %s%u", whence_s, to);
```

Comment:
This is not the right syntax for `addid`!

Reply:
sorry, copy and paste error from search code...
fixed with the last commit and I also centraliced the whence char handling